airflow/models/taskinstance.py

```diff
@@ -448,21 +448,30 @@ class TaskInstance(Base, LoggingMixin):
     note = association_proxy("task_instance_note", "content", creator=_creator_note)
     task: Operator  # Not always set...
 
-    def __init__(
-        self,
+    @staticmethod
+    def from_task(
```

Comment:
I like this, but this may not be viable. Creating a TaskInstance out-of-band has historically been used quite pervasively by users, and this could potentially break many installations. We could still do this (especially since we are trying to talk people off calling various Airflow internals anyway), but need to be aware of the possible implications.